app/src/main/java/com/example/recipeapp/DetailsActivity.java

```diff
@@ -54,5 +82,69 @@ protected void onCreate(Bundle savedInstanceState) {
             i.putExtra(DetailsActivity.INGREDIENTS, Parcels.wrap(ingredients));
             startActivity(i);
         });
+
+        binding.options.like.setOnClickListener(v -> viewModel.onLike());
+        binding.options.share.setOnClickListener(v -> onShare());
```

Comment:
you can pass recipe from here instead of passing it from activity, if needed, it will keep activity more stateless